### PR TITLE
Bug-fix: Streamlined Submit form button behavior on all forms

### PR DESCRIPTION
**Pull request by roshaans**
This PR addresses Issue #112 to prevent the user from pressing a form button twice. It also addresses a UX bug that keeps the form button disabled until the form is valid as well as changes have been made to it. No reason to keep the form button always in the clickable state when there are not any changes to be submitted. 

This behavior was changed on all flows using a form button for submission. 

**Solution**
- Take a similar approach to Near-Wallet by creating a reusable FormButton component
- Take advantage of Yup Resolver's isDirty & isValid state to enable/disable form buttons
- Create mutlisafe form Amount has a default value of 5. 

**Note**: The FormButton component comes with a few handy features that we can take advantage of in future PRs such as redirecting the user to a specific screen in the app upon the button being pressed, changing the text of the button while a form is being submitted.

### CODE REVIEW

File: src/ui/components/Main/MultiSafe/EditMultisafe/Form/Form.jsx [hunk 1]
```diff
@@ -41,15 +42,15 @@ export const Form = () => {
...
     const onSubmit = handleSubmit(async (data) => {
         if (await isBatchRequest({ data, history })) {
             setOpenConfirmModal(true);
             setFormData(data);
             return;
         }
-
-        onEditMultisafe({ data, history });
+        await onEditMultisafe({ data, history });
+        reset(data);
```
Comment (roshaans) [inline src/ui/components/Main/MultiSafe/EditMultisafe/Form/Form.jsx:53]: The use of `reset(data)` is to reset the `isDirty` property of the form so that once changes are made to the form after the form is submitted, the form button can be enabled again.

We do not necessarily need this unless we are sending the user back to the same page that they submitted the form on, but I imagine in the future we might. So it is there for precaution and does not hurt.

File: src/ui/components/Main/MultiSafe/NonFungibleTokens/Actions/TransferNFT.jsx [hunk 2]
```diff
@@ -7,20 +7,23 @@ import { useForm } from 'react-hook-form';
```
Comment (roshaans) [inline src/ui/components/Main/MultiSafe/NonFungibleTokens/Actions/TransferNFT.jsx:7]: I have tested all the flows in the app except the transfer NFTs flow. If anyone has NFTs in their multisafe, it'd be great to test that flow out.

**Review by marcinbodnar — CHANGES_REQUESTED** [review]
Great job @roshaans .

Two small things:
 - loading safe form - the button stays available
 - create multi safe form - there seems to be a small delay before the button becomes disabled

Also, it would be good to add a loader and change the text in the button to `Sending` (or something similar). We have a main loader, but it doesn't seem to work for example on editing safe action, adding the loader in the button will help. But it can be separate PR if you prefer.

**Comment by roshaans** [comment]
> * oading safe form - the button stays

Thank you for the review @marcinbodnar! I intentionally made the behaviors of the create and load multisafe forms different. I will try to address the issue for loading the mutlisafe in another PR that deals with account validation on blur. According to the logic used across the other forms, the form needs to be both in a valid state as well as changes need to be made to the form for the button to be enabled. But in its current state of the load multisafe form, the user is not given any validation of sorts if we keep the button disabled. 

As for the create multisafe form, I intentionally kept the create multisafe button enabled because the user will not get any sorts of feedback from the form that they need to `deposit` 5 Near into the mutlisafe until they press the create button and run into the issue. If I keep the form button disabled until everything is valid in the form, then any user who is in a hurry trying to create a vault would be pretty confused as to why the button won't turn into an enabled state unless they read the fine prints under one of the fields. 

I think we should by default keep the field at 5 Near. Then I can make it so I only turn the form button enabled if everything else is valid. 

TLDR; In a further PR #104, I am going to address validation on Blur. This will give the user instant feedback once they step out of a field. Then, I think it makes sense to get the behavior of these forms in line with the rest of the app.

**Comment by roshaans** [comment]
> Great job @roshaans .
> 
> Two small things:
> 
> * loading safe form - the button stays available
> * create multi safe form - there seems to be a small delay before the button becomes disabled
> 
> Also, it would be good to add a loader and change the text in the button to `Sending` (or something similar). We have a main loader, but it doesn't seem to work for example on editing safe action, adding the loader in the button will help. But it can be separate PR if you prefer.

As for the loader, I think we should tackle that in another PR as well. I tried to use Yup form's `isSubmitting` property to change the text on the form buttons, but this is not the best method. This is because there is a delay in the app sending you to the confirm transaction screen on the near wallet, and the actual 'form submission' is much faster than that so the interaction ends up looking very awkward. I think we will need a way to keep a global state object for pending actions like there is on the Near Wallet.

**Comment by roshaans** [comment]
> d tackle that in another PR as well. I tried to use Yup form's `isSubmitting` property to change the text on the form buttons, but this is not the best method. This is because there is a delay in the app sending you to the confirm transaction screen on the near wallet, and the actual 'form submission' is much faster than that so the interaction ends up looking very awkwa

U

> > * oading safe form - the button stays
> 
> Thank you for the review @marcinbodnar! I intentionally made the behaviors of the create and load multisafe forms different. I will try to address the issue for loading the mutlisafe in another PR that deals with account validation on blur. According to the logic used across the other forms, the form needs to be both in a valid state as well as changes need to be made to the form for the button to be enabled. But in its current state of the load multisafe form, the user is not given any validation of sorts if we keep the button disabled.
> 
> As for the create multisafe form, I intentionally kept the create multisafe button enabled because the user will not get any sorts of feedback from the form that they need to `deposit` 5 Near into the mutlisafe until they press the create button and run into the issue. If I keep the form button disabled until everything is valid in the form, then any user who is in a hurry trying to create a vault would be pretty confused as to why the button won't turn into an enabled state unless they read the fine prints under one of the fields.
> 
> I think we should by default keep the field at 5 Near. Then I can make it so I only turn the form button enabled if everything else is valid.
> 
> TLDR; In a further PR #104, I am going to address validation on Blur. This will give the user instant feedback once they step out of a field. Then, I think it makes sense to get the behavior of these forms in line with the rest of the app.

**Update:** I ended up making the necessary changes to get the form behavior of load/create multisafe screens in line with the rest of the app.

File: src/ui/components/Main/FormElements/FormButton/FormButton.jsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        color={color}
+        className={className}
+        disabled={disabled}
+        onClick={(e) => {
+            onClick && onClick(e);
+            linkTo && (linkTo.toLowerCase().startsWith('http') ? window.open(linkTo, '_blank') : history.push(linkTo));
+        }}
+    >
+        { 
+            sending ? (sendingString ? sendingString : 'sending') : children
```
Comment (Patrick1904) [inline src/ui/components/Main/FormElements/FormButton/FormButton.jsx:29]: Can we use the material ui loading/spinner button for loading state vs a custom `sendingString`? We've been meaning to do that in the wallet too since it's more standard than switching out the button text while loading:

https://mui.com/material-ui/react-button/

**Review by marcinbodnar — APPROVED** [review]
Thanks @roshaans LGTM

File: src/ui/components/Main/FormElements/FormButton/FormButton.jsx (hunk 3 above)
Comment (roshaans) [reply src/ui/components/Main/FormElements/FormButton/FormButton.jsx:29]: I will create a new a new ticket for this. We are using MUI v4, and the LoadingButton component is only available in v5. This will require us to setup a theming across the app so the components look as they do because in v5, it is not as simple as changing the css of the button as we currently do. Also, we do not want to have mutliple version of the library in one project. But I agree, the material-ui loading button is pretty clean.

Comment (Patrick1904) [reply src/ui/components/Main/FormElements/FormButton/FormButton.jsx:29]: Thanks for investigating! Not a blocker for this PR

**Review by Patrick1904 — APPROVED** [review]
